gateway/test/src/test/java/io/apiman/gateway/test/junit/GatewayRestTester.java

```diff
@@ -256,6 +256,7 @@ public void evaluate() throws Throwable {
                     }
     
                     testInfo.plan.runner.runTest(restTest, endpoint);
+                    gatewayServer.next(endpoint);
```

Comment:
This additional method greatly simplified things from my initial approaches (of which I had a few); it allows you to know when the next individual test is being run, so a reload can  be issued to the registry.